subsys/net/lib/nrf_cloud/src/nrf_cloud_fota_poll.c

```diff
@@ -153,6 +159,18 @@ static void http_fota_dl_handler(const struct fota_download_evt *evt)
 		LOG_ERR("FOTA download error: %d", evt->cause);
 
 		nrf_cloud_download_end();
+
+		/* Do an image reset if full modem FOTA fails. This is to avoid resuming the
+		 * download when doing a full modem FOTA, which does not currently work.
+		 */
+		if (ctx_ptr->img_type == DFU_TARGET_IMAGE_TYPE_FULL_MODEM) {
+			int err = fota_download_util_image_reset(DFU_TARGET_IMAGE_TYPE_FULL_MODEM);
```

Comment:
good idea... seems like this should be done in `fota_download` since it knows more about the specific failure mode and if it is possible to resume.

but if not, i believe it makes sense to do this only when `evt->cause` is one of these values:
```
	/** The update is invalid and was rejected. Retry will not help. */
	FOTA_DOWNLOAD_ERROR_CAUSE_INVALID_UPDATE,
	/** Actual firmware type does not match expected. Retry will not help. */
	FOTA_DOWNLOAD_ERROR_CAUSE_TYPE_MISMATCH,
	/** Generic error on device side. */
	FOTA_DOWNLOAD_ERROR_CAUSE_INTERNAL,
```

`FOTA_DOWNLOAD_ERROR_CAUSE_INTERNAL` seems to be a catch-all cause... it is possible there are cases when resuming might still work, but would take some investigation in `fota_download`.

Reply:
Hmm, the error code I get when it fails is FOTA_DOWNLOAD_ERROR_CAUSE_DOWNLOAD_FAILED.
I don't think resumption works at all for full modem fota atm. Sure, resumption does happen. But the hash is wrong after the image is downloaded.

Reply:
Created a ticket for this:
https://nordicsemi.atlassian.net/browse/NCSDK-29230